lib/charms/opensearch/v0/helper_storage.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""OpenSearch StorageResolver."""
+import logging
+from typing import TYPE_CHECKING, Literal, Optional
+
+from charms.opensearch.v0.constants_charm import (
+    AZURE_RELATION,
+    GCS_RELATION,
+    S3_RELATION,
+)
+from charms.opensearch.v0.models import (
+    AzureRelData,
+    DeploymentType,
+    GcsRelData,
+    ObjectStorageConfig,
+    S3RelData,
+)
+from pydantic import ValidationError
+
+if TYPE_CHECKING:
+    from charms.opensearch.v0.opensearch_base_charm import OpenSearchBaseCharm
+
+ObjectStorageType = Literal[
+    "s3", "azure", "gcs", "s3-pcluster", "azure-pcluster", "gcs-pcluster", "conflict"
+]
+
+logger = logging.getLogger(__name__)
+
+
+class ObjectStorageResolver:
+    """Resolves active object storage type and config from relations/peer-cluster."""
+
+    def __init__(self, charm: "OpenSearchBaseCharm") -> None:
+        """Initialize the object storage resolver object."""
+        self.charm = charm
+
+    def get_storage_type(self) -> Optional[ObjectStorageType]:  # noqa: C901
+        """Get the active object storage type from relations/peer-cluster."""
+        dep = self.charm.opensearch_peer_cm.deployment_desc()
```

Comment:
For consistency with the rest of the code base:
```suggestion
        deployment_desc = self.charm.opensearch_peer_cm.deployment_desc()
```

Reply:
Done.